libraries/AP_HAL_ChibiOS/RCOutput.cpp

```diff
@@ -737,7 +737,13 @@ void RCOutput::write(uint8_t chan, uint16_t period_us)
 
     chan -= chan_offset;
 
-    period[chan] = period_us;
+    if (corked) {
+        // when corked we put the updated period in a separate array which is
+        // copied to period[] when we push
+        period_corked[chan] = period_us;
```

Comment:
I mean shouldn't this always be the case? Couldn't you just assert on the non-corked case?

Reply:
there could be special cases where we write non-corked, for something going into 4.6.0 I wouldn't want to rule out non-corked output one some weird vehicle setup
one example would be ReplayGyroFFT
another in toy_mode test code
for 4.7.0 I'd be happy to enforce corked only, starting with SITL